### PR TITLE
Widgets fixes

### PR DESCRIPTION
**Pull request by Philip-Scott**

### CODE REVIEW

File: lib/Widgets/IncomingTransmissionList.vala [hunk 1]
```diff
@@ -101,8 +101,8 @@ public class Drop.Widgets.IncomingTransmissionList : Gtk.Box {
         }
 
         this.add (entry);
-
+        this.show_all ();
```
Comment (MarcusWichelmann) [inline lib/Widgets/IncomingTransmissionList.vala]: You removed the blank line accidently. Please avoid that.

File: lib/Widgets/IncomingTransmissionListEntry.vala [hunk 2]
```diff
@@ -40,7 +40,9 @@ public class Drop.Widgets.IncomingTransmissionListEntry : TransmissionListEntry
      * @param transmission Transmission to build the entry on.
      */
     public IncomingTransmissionListEntry (IncomingTransmission transmission) {
-        Object (transmission: transmission);
+        base ();
```
Comment (MarcusWichelmann) [inline lib/Widgets/IncomingTransmissionListEntry.vala]: Is that really needed!?

Comment (Philip-Scott) [inline lib/Widgets/IncomingTransmissionListEntry.vala]: Yes, that's what caused the parent's constructor to not run before, now im just forcing it to run

File: lib/Widgets/IncomingTransmissionList.vala (hunk 1 above)
Comment (Philip-Scott) [inline lib/Widgets/IncomingTransmissionList.vala]: it wasn't an accident, no need for that blank line

File: lib/Widgets/IncomingTransmissionListEntry.vala [hunk 3]
```diff
@@ -91,8 +93,7 @@ public class Drop.Widgets.IncomingTransmissionListEntry : TransmissionListEntry
         reject_button.clicked.connect (() => {
             try {
                 transmission.reject_transmission ();
-
-                accept_button.set_visible (false);
+                this.destroy ();
```
Comment (MarcusWichelmann) [inline lib/Widgets/IncomingTransmissionListEntry.vala:94]: Why are you destroying the widget here? Doesn't make any sense.

File: lib/Widgets/OutgoingTransmissionListEntry.vala [hunk 4]
```diff
@@ -37,9 +37,11 @@ public class Drop.Widgets.OutgoingTransmissionListEntry : TransmissionListEntry
      * @param transmission Transmission to build the entry on.
      */
     public OutgoingTransmissionListEntry (OutgoingTransmission transmission) {
-        Object (transmission: transmission);
+        base ();
+        this.transmission = transmission;
```
Comment (MarcusWichelmann) [inline lib/Widgets/OutgoingTransmissionListEntry.vala:40]: Please add a blank line to keep it consistent with IncomingTrnasmissionListEntry.vala

File: lib/Widgets/OutgoingTransmissionListEntry.vala [hunk 5]
```diff
@@ -47,6 +49,7 @@ public class Drop.Widgets.OutgoingTransmissionListEntry : TransmissionListEntry
     private void build_ui () {
         cancel_button = new Gtk.Button.from_icon_name ("process-stop-symbolic", Gtk.IconSize.BUTTON);
         cancel_button.valign = Gtk.Align.CENTER;
+        action_area.add (cancel_button);
```
Comment (MarcusWichelmann) [inline lib/Widgets/OutgoingTransmissionListEntry.vala:52]: add a blank line before of this line.

File: lib/Widgets/IncomingTransmissionListEntry.vala (hunk 3 above)
Comment (Philip-Scott) [inline lib/Widgets/IncomingTransmissionListEntry.vala:94]: if you clicked reject, widget is no longer needed and should disappear from the Transmission List.

File: lib/Widgets/OutgoingTransmissionListEntry.vala [hunk 6]
```diff
@@ -112,9 +116,7 @@ public class Drop.Widgets.OutgoingTransmissionListEntry : TransmissionListEntry
 
             if (files.length == 0) {
                 warning ("File list invalid.");
-
```
Comment (MarcusWichelmann) [inline lib/Widgets/OutgoingTransmissionListEntry.vala:115]: revoke please.

File: lib/Widgets/OutgoingTransmissionListEntry.vala [hunk 7]
```diff
@@ -112,9 +116,7 @@ public class Drop.Widgets.OutgoingTransmissionListEntry : TransmissionListEntry
...
                 this.destroy ();
-
```
Comment (MarcusWichelmann) [inline lib/Widgets/OutgoingTransmissionListEntry.vala:117]: revoke please.

File: lib/Widgets/TransmissionListEntry.vala [hunk 8]
```diff
@@ -41,10 +41,11 @@ public class Drop.Widgets.TransmissionListEntry : Gtk.Grid {
      * Creates a new transmission entry and it's ui.
      */
     public TransmissionListEntry () {
-        build_ui ();
+        this.build ();
```
Comment (MarcusWichelmann) [inline lib/Widgets/TransmissionListEntry.vala]: is it really needed to rename the method to build() ? They are all private, I doup that's neccessary.

File: lib/Widgets/IncomingTransmissionListEntry.vala (hunk 2 above)
Comment (MarcusWichelmann) [inline lib/Widgets/IncomingTransmissionListEntry.vala]: Strange. Have you tested it without base(); I doubt that's required. ;)

File: lib/Widgets/IncomingTransmissionListEntry.vala (hunk 3 above)
Comment (MarcusWichelmann) [inline lib/Widgets/IncomingTransmissionListEntry.vala:94]: ups, sorry, I thought that's the accept_button. :smile:

**Comment by MarcusWichelmann** [comment]
Please avoid doing codestyle changes in this commit. Im thinking about every blank line and there is a reason why Im adding them :smile:

File: lib/Widgets/TransmissionListEntry.vala [hunk 9]
```diff
@@ -57,6 +57,7 @@ public class Drop.Widgets.TransmissionListEntry : Gtk.Grid {
 
         secondary_label = new Gtk.Label ("");
         secondary_label.halign = Gtk.Align.START;
+        secondary_label.hexpand = true;
```
Comment (Philip-Scott) [inline lib/Widgets/TransmissionListEntry.vala:60]: Moved hexpand to the secondary label to:
- Avoid buttons to expanding more than needed (Causing un-uniformity) 
- didn't add it to primary lable to not break the elipsize

File: lib/Widgets/TransmissionListEntry.vala [hunk 10]
```diff
@@ -41,7 +41,7 @@ public class Drop.Widgets.TransmissionListEntry : Gtk.Grid {
      * Creates a new transmission entry and it's ui.
      */
     public TransmissionListEntry () {
-        build_ui ();
+        this.build_ui ();
```
Comment (MarcusWichelmann) [inline lib/Widgets/TransmissionListEntry.vala]: Is this. here needed? There is only one build_ui accessible here.

**Comment by MarcusWichelmann** [comment]
Thank you, looks good now!

**Pull request merged**